File: clientlibs/js/src/functions/log.ts
import { UpGradeClientEnums, UpGradeClientInterfaces } from '../types';
import fetchDataService from '../common/fetchDataService';
import { ILogInput } from 'upgrade_types';

export default async function log(
  httpClient: UpGradeClientInterfaces.IHttpClientWrapper,
  url: string,
  userId: string,
  token: string,
  clientSessionId: string,
  value: ILogInput[],
  sendAsAnalytics = false
): Promise<UpGradeClientInterfaces.ILog[]> {
  const data = {
    userId,
    value,
  };
  const logResponse = await fetchDataService(
    httpClient,
    url,
    token,
    clientSessionId,
    data,
    UpGradeClientEnums.REQUEST_TYPES.POST,
    sendAsAnalytics
  );
  if (logResponse.status) {
    return logResponse.data;
  } else {
    throw new Error(JSON.stringify(logResponse.message));
  }
}

File: clientlibs/csharp/src/functions/Log.cs
using System.Collections.Generic;
using System.Threading.Tasks;
using Enums;
using Interfaces;
using Newtonsoft.Json;

public static class LogClass
{
  public static async Task<Log[]> Log(
    string url,
    string userId,
    string token,
    string clientSessionId,
    LogInput[] value,
    bool sendAsAnalytics = false)
  {
    var data = new
    {
      userId,
      value
    };
    var response = await FetchDataServiceClass.FetchDataService(url, token, clientSessionId, data, REQUEST_TYPES.POST, sendAsAnalytics);
    if (response.status)
    {
      var responseData = JsonConvert.DeserializeObject<Log[]>(response.data.ToString());
      return responseData as Log[];
    }
    else
    {
      throw new Exception(JsonConvert.SerializeObject(response.message));
    }
  }
}
